src/platform/web/ui/login/LoginView.js

```diff
@@ -16,61 +16,49 @@ limitations under the License.
-        });
-        
         return t.div({className: "PreSessionScreen"}, [
+            t.a({className: "button-utility LoginView_back", href: vm.cancelUrl}),
             t.div({className: "logo"}),
-            t.div({className: "LoginView form"}, [
-                t.h1([vm.i18n`Sign In`]),
-                t.if(vm => vm.error, t => t.div({className: "error"}, vm => vm.error)),
-                t.form({
-                    onSubmit: evnt => {
-                        evnt.preventDefault();
-                        vm.login(username.value, password.value, homeserver.value);
-                    }
-                }, [
-                    t.div({className: "form-row"}, [t.label({for: "username"}, vm.i18n`Username`), username]),
-                    t.div({className: "form-row"}, [t.label({for: "password"}, vm.i18n`Password`), password]),
-                    t.div({className: "form-row"}, [t.label({for: "homeserver"}, vm.i18n`Homeserver`), homeserver]),
-                    t.mapView(vm => vm.loadViewModel, loadViewModel => loadViewModel ? new SessionLoadStatusView(loadViewModel) : null),
-                    t.div({className: "button-row"}, [
-                        t.a({
-                            className: "button-action secondary",
-                            href: vm.cancelUrl
-                        }, [vm.i18n`Go Back`]),
-                        t.button({
-                            className: "button-action primary",
-                            type: "submit"
-                        }, vm.i18n`Log In`),
-                    ]),
-                ]),
-                // use t.mapView rather than t.if to create a new view when the view model changes too
-                t.p(hydrogenGithubLink(t))
-            ])
+            t.h1([vm.i18n`Sign In`]),
+            t.mapView(vm => vm.completeSSOLoginViewModel, vm => vm ? new CompleteSSOView(vm) : null),
+            t.if(vm => vm.showHomeserver, (t, vm) => t.div({ className: "LoginView_sso form form-row" },
+                [
+                    t.label({for: "homeserver"}, vm.i18n`Homeserver`),
+                    t.input({
+                        id: "homeserver",
+                        type: "text",
+                        placeholder: vm.i18n`Your matrix homeserver`,
+                        value: vm.defaultHomeServer,
+                        disabled: vm => vm.isBusy,
+                        onChange: event => vm.updateHomeServer(event.target.value),
+                    })
+                ]
+            )),
+            t.mapView(vm => vm.passwordLoginViewModel, vm => vm ? new PasswordLoginView(vm): null),
+            t.if(vm => vm.passwordLoginViewModel && vm.startSSOLoginViewModel, t => t.p({className: "LoginView_separator"}, vm.i18n`or`)),
+            t.mapView(vm => vm.startSSOLoginViewModel, vm => vm ? new StartSSOLoginView(vm) : null),
+            t.if(vm => vm.errorMessage, (t, vm) => t.p({className: "LoginView_error"}, vm.i18n(vm.errorMessage))),
```

Comment:
You can add the error class for redness and boldness:
```suggestion
            t.if(vm => vm.errorMessage, (t, vm) => t.p({className: "LoginView_error error"}, vm.i18n(vm.errorMessage))),
```

Reply:
Perhaps put this at the top as well, before the homeserver field?